### PR TITLE
Analog: new test for agc3

### PR DESCRIPTION
**Pull request by cmrincon**
## Description
At present agc3 tests are performed with a simple sinusoidal wave which makes easy for the agc block to track input and adjust gain.

In this PR i suggest the use of a AM modulated signal with m=0.99 and  mean amplitude =1. This change will likely make the agc struggle. Also this test can be extended to agc and agc2.

Compared to previous tests, delta error values have been decreased to 0.1 to adjust the difficulty of the test properly.

Finally, i noticed that a mean amplitude different to 1 makes the AGC dramatically fail the test and perform horrible.
 

## Related Issue
[ AGC3: needs proper testing #6356](https://github.com/gnuradio/gnuradio/issues/6356)

## Which blocks/areas does this affect?
Affects to the agc3 tests.

## Testing Done
I changed different AM modulation parameters and different decay & attack rates, choosing the ones which make the agc perform better.

## Checklist

- [X] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [X] I have squashed my commits to have one significant change per commit. 
- [X] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [X] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Comment by willcode** [comment]
Thanks for resubmitting - glad the CI is happy this time.

File: gr-analog/python/analog/qa_agc.py [hunk 1]
```diff
@@ -492,6 +492,39 @@ def test_006_003_agc3_setter_errors(self):
             with self.assertRaises(ValueError):
                 troublemaker(-3.0)
 
+    def test_006_004_agc3(self, stride=1):
```
Comment (mbr0wn) [inline gr-analog/python/analog/qa_agc.py:495]: Can you please give this either a more descriptive name, or add a docstring to explain what this does?

**Review by mbr0wn — CHANGES_REQUESTED** [review]

**Review by mbr0wn — APPROVED** [review]
I'm not sure if @marcusmueller agrees this gives us the level of testing we need, but the change itself is OK to go into main.

**Comment by willcode** [comment]
Hard to design tests for dynamic blocks without copying the logic or getting a reference vector from the block and making sure it doesn't change in the future.

**Pull request merged**